src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/SupplierAssessmentController.kt

```diff
@@ -0,0 +1,42 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.controller
+
+import org.springframework.http.HttpStatus
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken
+import org.springframework.web.bind.annotation.PatchMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.server.ResponseStatusException
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.authorization.UserMapper
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.SupplierAssessmentDTO
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.UpdateAppointmentDTO
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.ReferralService
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.SupplierAssessmentService
+import java.util.UUID
+
+@RestController
+class SupplierAssessmentController(
+  private val supplierAssessmentService: SupplierAssessmentService,
+  private val referralService: ReferralService,
+  private val userMapper: UserMapper,
+) {
+  @PatchMapping("/sent-referral/{id}/supplier-assessment-appointment")
```

Comment:
this seems an odd URL to have outside of the referrals controller?

how come this isn't updating the supplier assessment by id?

i would expect this to be more like `POST /supplier-assessment/{id}/appointment`

Reply:
thinking out loud, about suppler assessment appointments, we need to

- book them
- reschedule them (re-book)
- record attendance, behaviour and feedback
- and list them

all of these are in the context of the referral

Since we already have `GET /action-plan?referralId=...` pattern, I think it's ok to have

- `GET /supplier-assessment?referralId=...`
- `POST /supplier-assessment` with `referralId`
- `PATCH /supplier-assessment/{id}`

It would be really good if we could mention the key domain verbs here somehow -- `book`, `attendance`, etc.

Reply:
the endpoints that were going to be created (according to contract) were:

get saa
patch saa
record saa attendance
record saa behaviour
submit saa session feedback.

So, following the above, would you like these to be:
GET: /supplier-assessment-appointment/{id}
PATCH: /supplier-assessment-appointment/{id}
POST: /supplier-assessment-appointment/{id}/record-attendance
POST: /supplier-assessment-appointment/{id}/record-behaviour
POST: /supplier-assessment-appointment/{id}/submit

Reply:
Decision from meeting:

supplier assessment URLs
# ReferralsController
GET /referrals/{referralId}/supplier-assessment -> {id: …, appointments: [], currentAppointmentId: … } (SupplierAssessmentDTO) = get supplier assessment for referral
# SupplierAssessmentsController
PUT /supplier-assessment/{supplierAssessmentId}/schedule-appointment -> AppointmentDTO = book or reschedule appointment
	- if there is no appointment, book a new one.
	- if there is an appointment with no attendance data, reschedule
	- if there is an appointment with attendance but didn’t attend, book new one
	- if there is an appointment with attendance but did attend, error
# AppointmentsController
PUT /appointments/{id}/record-attendance - > AppointmentDTO
PUT /appointments/{id}/record-behaviour - > AppointmentDTO
POST /appointments/{id}/submit-feedback -> AppointmentDTO